tsfresh/feature_extraction/extraction.py

```diff
@@ -208,28 +207,31 @@ def _do_extraction(df, column_id, column_value, column_kind,
     :param disable_progressbar: Do not show a progressbar while doing the calculation.
     :type disable_progressbar: bool
 
-    :param distributor: Advanced parameter: set this to an instance name that you want to use as a
-             distributor. See the utilities/distribution.py for more information. Leave to None, if you want
-             TSFresh to choose the best distributor.
-    :type distributor: Distributor
+    :param distributor: Advanced parameter:  See the utilities/distribution.py for more information.
+                         Leave to None, if you want TSFresh to choose the best distributor.
+    :type distributor: DistributorBaseClass
 
     :return: the extracted features
     :rtype: pd.DataFrame
     """
     data_in_chunks = [x + (y,) for x, y in df.groupby([column_id, column_kind])[column_value]]
 
     if distributor is None:
+
         if n_jobs == 0:
-            distributor_class = MapDistributor
+            distributor = MapDistributor(disable_progressbar=disable_progressbar,
+                                         progressbar_title="Feature Extraction")
         else:
-            distributor_class = MultiprocessingDistributor
+            distributor = MultiprocessingDistributor(n_workers=n_jobs, disable_progressbar=disable_progressbar,
+                                                     progressbar_title="Feature Extraction")
 
-        distributor = distributor_class(n_workers=n_jobs,
-                                        disable_progressbar=disable_progressbar,
-                                        progressbar_title="Feature Extraction")
+    if not isinstance(distributor, DistributorBaseClass):
```

Comment:
do we really want to be that strict? If a user implements everything that is needed, it is fine to have his own base class (he is of course responsible on his own).
Because of duck typing, it is barely needed to check for a type in python....

Reply:
I want to catch stupid errors (e.g. user forgetting to instantiate the distributor object) with that strict class check.

I do not think that a user will be able to implement a custom distributor without inheriting from the base class. To write the distributor properly, you really need to understand how tsfresh uses it.

Also, there are not too many cluster frameworks, so also the number of possible distributors is limited. 

All in all, I think this motivates the strict test instead of duck typing.